assets/RNAseq_report.Rmd

```diff
@@ -1780,6 +1785,180 @@ pg[['x']][['layout']][['annotations']][[1]][['y']] <- -0.05
 config(pg, modeBarButtonsToAdd = list(svg_exp)) %>% layout(margin = list(l=80, b = ifelse(length(allgenes_files)>2, 150, 100)), width=700, height = 500*length(allgenes_files))
 ```
 
+
+```{r enhanced_volcano, echo=FALSE, message=FALSE, warning=FALSE}
+
+library(EnhancedVolcano)
+library(ggplot2)
+
+theme_set(theme_bw(base_family = "ArialMT") +
+theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), text = element_text(family="ArialMT")))
+
+
+thousandFormat <- function(x){
+  formatC(x, format='f', big.mark=',', digits=0)
+  }
+
+allgenes_files <- list.files(path=paste0(wd, "/differential_gene_expression/allgenes/"))
+volcano_list <-  c()
+contrast_list <- c()
+
+
+for (file in allgenes_files){
+
+    # Reading DE genes list
+    fname <- tools::file_path_sans_ext(basename(file))
+    DE_genes <- read.csv(file = paste0(wd, "/differential_gene_expression/allgenes/", file), sep="\t", header = T)
+    DE_genes$contrast <- rep(fname, nrow(DE_genes))
+    contrast_len <- sapply(strsplit(fname, '_'), length)
+    # Rename Contrast for plot
+    # DE_contrast_condition_conditionName_conditionA_vs_conditionB --> conditionA vs conditionB
+    contrast_list <- c(contrast_list, sapply(strsplit(fname, '_'), function(x) paste(x[5:contrast_len], collapse=' ')))
+
+    # label: gene_name
+    # x: log2FoldChange
+    # y: -log10(padj)
+
+    DE_genes <- na.omit(DE_genes)
+
+    # convert zeros to small value (as in EnhancedVolcano)
+    DE_genes[which(DE_genes[['padj']] == 0), 'padj'] <- min(
+    DE_genes[which(DE_genes[['padj']] != 0), 'padj'], na.rm = TRUE) * 10^-1
```

Comment:
I find the newline not intuitive, either all in one line or with indent?
But why the min non-zero padj * 10⁻1, isnt that arbitrary? But anything else would be arbitrary as well I guess.
What happens with the genes that have padj with NA? Are they just omitted? (Fine if yes, just needs to be documented in the reporting if it isnt)

Reply:
Thanks for the feedback @d4straub!
This handling of p-values was inspired by the EnhancedVolcano package. 
I will change the formatting to a one-liner and document this processing step. 

Concerning NA values: They are omitted as in the last releases of the rnadeseq pipeline and I applied no changes to the code. I can add documentation on that in a new PR that is not related to the volcano plot.